src/dispatch/plugins/dispatch_slack/service.py

```diff
@@ -281,13 +286,18 @@ def send_message(
         blocks=blocks,
     )
 
+    domain = make_call(
+        client,
+        SlackAPIGetEndpoints.team_info
+    )["team"]["domain"]
+
     if persist:
         add_pin(client, response["channel"], response["ts"])
 
     return {
         "id": response["channel"],
         "timestamp": response["ts"],
-        "weblink": f"https://slack.com/app_redirect?channel={response['id']}",  # TODO should we fetch the permalink?
+        "weblink": f"https://{domain}.slack.com/app_redirect?channel={response['id']}",  # TODO should we fetch the permalink?
```

Comment:
I don't think fetching the domain is necessary for sending or updating messages. AFAIR we're not using the response for anything. If we did, we should use this endpoint to get the permalink to the message instead: https://api.slack.com/methods/chat.getPermalink